Statement/Statement.php

```diff
@@ -161,52 +167,48 @@ public function getAccountNumber()
         return $this->accountNumber;
     }
 
-    /**
-     * @param $accountNumber
-     *
-     * @return $this
-     */
-    public function setAccountNumber($accountNumber)
+    public function setAccountNumber($accountNumber): self
     {
         $this->accountNumber = $accountNumber;
 
-        return $this;
-    }
-
-    /**
-     * Split account number to parts
-     *
-     * @return array
-     */
-    public function getParsedAccountNumber()
-    {
-        $parsedAccountNumber = array(
+        $this->parsedAccountNumber = [
             'prefix'   => null,
             'number'   => null,
             'bankCode' => null
-        );
-
-        $accountNumber = $this->getAccountNumber();
+        ];
 
         $splitBankCode = explode('/', $accountNumber);
         if (count($splitBankCode) === 2) {
-            $parsedAccountNumber['bankCode'] = $splitBankCode[1];
+            $this->parsedAccountNumber['bankCode'] = $splitBankCode[1];
         }
 
+        // Support format with separator or without separator
         $splitNumber = explode('-', $splitBankCode[0]);
+        $firstPart = $splitNumber[0];
         if (count($splitNumber) === 2) {
-            $parsedAccountNumber['prefix'] = $splitNumber[0];
-            $parsedAccountNumber['number'] = $splitNumber[1];
+            $this->parsedAccountNumber['prefix'] = $firstPart;
+            $this->parsedAccountNumber['number'] = $splitNumber[1];
         } else {
-            if (strlen($splitNumber[0]) <= 10) {
-                $parsedAccountNumber['number'] = $splitNumber[0];
+            $firstPartLength = strlen($firstPart);
+            if ($firstPartLength <= self::BankAcountNumberMaxLength) {
+                $this->parsedAccountNumber['number'] = $firstPart;
             } else {
-                $parsedAccountNumber['prefix'] = substr($splitNumber[0], 0, strlen($splitNumber[0]) - 10);
-                $parsedAccountNumber['number'] = substr($splitNumber[0], -10, 10);
+                $this->parsedAccountNumber['prefix'] = substr($firstPart, 0, $firstPartLength - self::BankAcountNumberMaxLength);
+                $this->parsedAccountNumber['number'] = substr($firstPart, -self::BankAcountNumberMaxLength, self::BankAcountNumberMaxLength);
             }
         }
 
-        return $parsedAccountNumber;
+        return $this;
+    }
+
+    /**
+     * Split account number to parts
+     *
+     * @return array{prefix: ?string, number: ?string, bankCode: ?string}
+     */
+    public function getParsedAccountNumber(): array
```

Comment:
Previously this method was constructing `$parsedAccountNumber` on every transactions - this can lead to bad performance - constructing is done in `setAccountNumber`.